app/src/main/jni/modules/recorder.cpp

```diff
@@ -26,6 +28,41 @@ struct RecordingModule : public AlgorithmModule {
                 (videoRecName.is_null() || !recordCamera) ? "" : videoRecName.get<std::string>());
 
         recorder->setVideoRecordingFps(settings.at("targetFps").get<float>());
+
+        infoFile = !settings.at("infoFileName").is_null() ? settings.at("infoFileName") : "";
+        paramsFile = !settings.at("parametersFileName").is_null() ? settings.at("parametersFileName") : "";
+    }
+    
+    void writeInfoFile(const std::string &mode, const std::string &device) {
+        // TODO: Move implementation to recorder?
+        if (infoFile.empty())
+            return;
+        json infoJson = R"(
+            {
+                "camera mode": "",
+                "device": "",
+                "tags": [
+                    "android"
+                ]
+            }
+        )"_json;
+        infoJson["camera mode"] = mode;
+        infoJson["device"] = device;
+        std::ofstream fileOutput(infoFile);
+        std::ostream &output(fileOutput);
+        output << infoJson.dump() << std::endl;
+    }
+
+    void writeParamsFile(float focalLength) {
+        // TODO: Move implementation to recorder?
+        if (paramsFile.empty())
+            return;
+        std::ofstream fileOutput(paramsFile);
+        std::ostream &output(fileOutput);
+        output << "imuToCameraMatrix -0,-1,0,-1,0,0,0,0,-1;focalLength "
+               << std::to_string(focalLength)
+               << ";"
+               << std::endl;
```

Comment:
If recording while running odometry, it would be nice if the custom parameters defined in our `module.cpp` implementation got appended here. At least I think the original idea was that the `parameters.txt` would record all the settings needed to reproduce the session offline, and that's how the iOS version is implemented. However, in practice I've rarely made use of that capability — usually we test and record separately.

The reproducibility feature would also mean `focalLength` should be written here only if a constant focal length was set for duration of the recording.